### ISSUE TITLE
[Home fragment] Tap triggers refresh when weather is null

### ISSUE THREAD

**Issue by min7-i** [breezy-weather/breezy-weather#1380]
### Steps to reproduce

1. Search for a new location.
2. Enable flight mode.
3. Add the new location.
4. Tap on the new location to open the main screen (home fragment).
5. Tap on the empty screen.

Alternatively, you can add your current location and disable location search.

### Expected behavior

A single tap does not trigger a weather refresh.

### Actual behavior

A single tap triggers a weather refresh.

https://github.com/user-attachments/assets/844bbcdd-0e71-4a79-9535-92a202928522

### Weather source used

N/A

### Breezy Weather version

5.2.8

### Android version

Android 15

### Device

Google Pixel 6a

### Other details

Tap-to-refresh is triggered by a [touch listener](https://github.com/breezy-weather/breezy-weather/blob/0ba8f8575f97ee7ffd0bf5e450b2657d79a05772/app/src/main/java/org/breezyweather/main/fragments/HomeFragment.kt#L353-L363) which is added in the home fragment when no weather data is available for a location.

In my opinion, that behavior is rather unintiuitive. Additionally, it also conflicts with swipes to switch locations as a tap triggers a refresh while switching to a different location. As a result, I'd suggest to remove the tap-to-refresh listener.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> In my opinion, that behavior is rather unintiuitive

Let's flip the chessboard: is a "pull to refresh" intuitive in that situation?

Let's take the most common case of "empty weather": someone who just installed Breezy Weather, only added current location, and it failed to refresh. They probably don't know (yet) about the pull to refresh. I believe that's the reason why it was added (additionally, back in Geometric Weather times, current location was automatically added as your first location, so it was way more common).

**Comment by min7-i** [comment]
> They probably don't know (yet) about the pull to refresh.

Agreed. I'd suggest to add a text in the center of the home fragment when weather is null, e.g. "Pull down to refresh weather data". That way users are made aware of the feature and we don't need the "tap to refresh".

**Comment by papjul** [comment]
LGTM. Must take into account the ~10 sec delay before retrying (a different message like "Please wait before retrying").

**Comment by min7-i** [comment]
I won't be able to work on it for now, so feel free to flag it as "Open to contributions".
